packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -541,21 +541,38 @@ Future<XcodeBuildResult> buildXcodeProject({
 }
 
 void updateShorebirdYaml(BuildInfo buildInfo, BuildableIOSApp app) {
+  final String resolvedAppName = app.name ?? 'Runner.app';
   final File shorebirdYaml = globals.fs.file(
     globals.fs.path.join(
       app.archiveBundleOutputPath,
       'Products',
       'Applications',
-      app.name ?? 'Runner.app',
+      resolvedAppName,
       'Frameworks',
       'App.framework',
       'flutter_assets',
       'shorebird.yaml',
     ),
   );
   if (!shorebirdYaml.existsSync()) {
+    // Find the closest existing parent of the file.
+    Directory parent = shorebirdYaml.parent;
+
+    int i = 0;
+    // The max depth is just a hard limit to prevent the cli from going to far behind.
```

Comment:
```suggestion
    // The max depth is just a hard limit to prevent the cli from going too far back in the
    // folder tree and unintentionally "invading" a user folder that isn't the project. 
    //
```